crates/transport_iroh/tests/harness.rs

```diff
@@ -0,0 +1,190 @@
+    pub new_listening_address: Arc<dyn Fn(Url) + 'static + Send + Sync>,
+    /// Mock function to implement [`TxBaseHandler::peer_connect()`]
+    pub peer_connect: Arc<dyn Fn(Url) -> K2Result<()> + 'static + Send + Sync>,
+    /// Mock function to implement [`TxBaseHandler::peer_disconnect()`]
+    pub peer_disconnect:
+        Arc<dyn Fn(Url, Option<String>) + 'static + Send + Sync>,
+    /// Mock function to implement [`TxHandler::preflight_gather_outgoing()`]
+    pub preflight_gather_outgoing:
+        Arc<dyn Fn(Url) -> K2Result<bytes::Bytes> + 'static + Send + Sync>,
+    /// Mock function to implement [`TxHandler::preflight_validate_incoming()`]
+    pub preflight_validate_incoming:
+        Arc<dyn Fn(Url, bytes::Bytes) -> K2Result<()> + 'static + Send + Sync>,
+    /// Mock function to implement [`TxSpaceHandler::recv_space_notify()`]
+    pub recv_space_notify: Arc<
+        dyn Fn(Url, SpaceId, bytes::Bytes) -> K2Result<()>
+            + 'static
+            + Send
+            + Sync,
+    >,
+    /// Mock function to implement [`TxModuleHandler::recv_module_msg()`]
+    pub recv_module_msg: Arc<
+        dyn Fn(Url, SpaceId, String, bytes::Bytes) -> K2Result<()>
+            + 'static
+            + Send
+            + Sync,
+    >,
+    /// Mock function to implement [`TxSpaceHandler::set_unresponsive()`]
+    pub set_unresponsive:
+        Arc<dyn Fn(Url, Timestamp) -> K2Result<()> + 'static + Send + Sync>,
+    /// Mock function to implement [`TxSpaceHandler::are_all_agents_at_url_blocked()`]
+    #[allow(clippy::type_complexity)]
+    pub are_all_agents_at_url_blocked:
+        Arc<dyn Fn(&Url) -> K2Result<bool> + 'static + Send + Sync>,
+    /// The current URL of this peer.
+    pub current_url: Arc<Mutex<Url>>,
+}
+
+impl std::fmt::Debug for MockTxHandler {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("MockTxHandler").finish()
+    }
+}
+
+impl Default for MockTxHandler {
+    fn default() -> Self {
+        Self {
+            new_listening_address: Arc::new(|_| {}),
+            peer_connect: Arc::new(|_| Ok(())),
+            peer_disconnect: Arc::new(|_, _| {}),
+            preflight_gather_outgoing: Arc::new(|_| Ok(bytes::Bytes::new())),
+            preflight_validate_incoming: Arc::new(|_, _| Ok(())),
+            recv_space_notify: Arc::new(|_, _, _| Ok(())),
+            recv_module_msg: Arc::new(|_, _, _, _| Ok(())),
+            set_unresponsive: Arc::new(|_, _| Ok(())),
+            are_all_agents_at_url_blocked: Arc::new(|_| Ok(false)),
+            current_url: Arc::new(Mutex::new(
+                Url::from_str("http://url.not.set:0/0").unwrap(),
+            )),
+        }
+    }
+}
+
+impl TxBaseHandler for MockTxHandler {
+    fn new_listening_address(&self, url: Url) -> BoxFut<'static, ()> {
+        (self.new_listening_address)(url.clone());
+        *self.current_url.lock().unwrap() = url;
```

Comment:
The handler has a `current_url` field which gets updated by default when a new address is assigned.